lib/ansible/modules/source_control/git.py

```diff
@@ -1117,8 +1111,13 @@ def main():
         result['warnings'].append("Your git version is too old to fully support the depth argument. Falling back to full checkouts.")
         depth = None
 
-    recursive = module.params['recursive']
-    track_submodules = module.params['track_submodules']
+    if submodule_depth is not None and git_version_used < LooseVersion('1.8.4'):
+        result['warnings'].append("Your git version is too old to fully support the submodule_depth argument. Falling back to full checkouts.")
```

Comment:
@ju2wheels normally, I insist on implementing a fallback implementation instead of giving up, like https://github.com/ansible/ansible/pull/41712/files#diff-423dabf0f8d841053684543485e386c9R487

Reply:
@webknjaz Please re-review, this does not give up and by resetting `submodule_depth` to `None` here, when we call the `submodule_update` method, it causes the `--depth` parameter not to be passed to `git` CLI (https://github.com/ansible/ansible/pull/49000/files/f70ead2057717155849fc2c41bc7a8cb8d71e040#diff-423dabf0f8d841053684543485e386c9R838) with the value that was previously set in `submodule_depth`, therefore falling back to a full clone as stated in the warning.

Reply:
ping @webknjaz